reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt

```diff
@@ -1,27 +1,50 @@
 package com.badoo.reaktive.test.base
 
+import com.badoo.reaktive.utils.printStack
 import kotlin.test.assertEquals
 import kotlin.test.assertFalse
 import kotlin.test.assertNotNull
 import kotlin.test.assertNull
 import kotlin.test.assertTrue
+import kotlin.test.fail
 
 fun TestObserver.assertError() {
     assertTrue(isError, "Source did not fail")
 }
 
 fun TestObserver.assertError(expectedError: Throwable) {
-    assertEquals(expectedError, error, "Source error does not match")
+    val error = error
+    if (error == null) {
```

Comment:
You can beautify it a little. But up to you, `when` can be overkill here and I don't know, will it be optimised by compiler.

```kotlin
when (val error = error) {
     null -> fail()
     else -> { ... }
}
```

Reply:
Personally I don't like replacing `if-else` with `when` :-)

Reply:
Personally I don't like replacing `if-else` with `when`, let's keep it like this?

Reply:
Keep it.